File: myschedule-web/src/main/resources/myschedule/script/examples/calendars.rb
# JRuby Examples

# Create Quartz Calendar objects
require 'java'
java_import Java::OrgQuartzImplCalendar::CronCalendar
java_import Java::MyscheduleQuartzExtraJob::LoggerJob
java_import Java::MyscheduleQuartzExtra::SchedulerTemplate

cal = CronCalendar.new('* * * * JAN ?')
scheduler.add_calendar('SkipJan', cal, true, false)
cal = CronCalendar.new('* * * ? * SAT,SUN')
scheduler.add_calendar('SkipWeekEnd', cal, true, false)
cal = CronCalendar.new('* * 12-13 * * ?')
scheduler.add_calendar('SkipLunch', cal, true, false)

# Create jobs that uses the calendars.
job = SchedulerTemplate.create_job_detail('CalJob', LoggerJob.java_class)
trigger = SchedulerTemplate.create_cron_trigger('CalJob1', '0 0 * * * ?')
trigger.set_calendar_name('SkipWeekEnd')
scheduler.schedule_job(job, trigger)

job = SchedulerTemplate.create_job_detail('CalJob2', LoggerJob.java_class)
trigger = SchedulerTemplate.create_cron_trigger('CalJob2', '0 0 * * * ?')
trigger.set_calendar_name('SkipJan')
scheduler.schedule_job(job, trigger)

job = SchedulerTemplate.create_job_detail('CalJob3', LoggerJob.java_class)
trigger = SchedulerTemplate.create_cron_trigger('CalJob3', '0 0 * * * ?')
trigger.set_calendar_name('SkipLunch')
scheduler.schedule_job(job, trigger)

File: myschedule-web/src/main/resources/myschedule/script/examples/calendars.js
// JavaScript Examples

//Create Quartz Calendar objects
importPackage(Packages.org.quartz.impl.calendar);
importClass(Packages.myschedule.quartz.extra.job.LoggerJob);

cal = CronCalendar('* * * * JAN ?');
scheduler.addCalendar('SkipJan', cal, true, false);
cal = CronCalendar('* * * ? * SAT,SUN');
scheduler.addCalendar('SkipWeekEnd', cal, true, false);
cal = CronCalendar('* * 12-13 * * ?');
scheduler.addCalendar('SkipLunch', cal, true, false);

// Create jobs that uses the calendars.
job = scheduler.createJobDetail('CalJob', LoggerJob);
trigger = scheduler.createCronTrigger('CalJob1', '0 0 * * * ?');
trigger.setCalendarName('SkipWeekEnd');
scheduler.scheduleJob(job, trigger);

job = scheduler.createJobDetail('CalJob2', LoggerJob);
trigger = scheduler.createCronTrigger('CalJob2', '0 0 * * * ?');
trigger.setCalendarName('SkipJan');
scheduler.scheduleJob(job, trigger);

job = scheduler.createJobDetail('CalJob3', LoggerJob);
trigger = scheduler.createCronTrigger('CalJob3', '0 0 * * * ?');
trigger.setCalendarName('SkipLunch');
scheduler.scheduleJob(job, trigger);
